### PR TITLE
Move to Shared - SqlTransaction

### PR DESCRIPTION
**Pull request by Kaur-Parminder**
This is part of #1261 Code merge effort.  Move the common code to shared folder and to avoid complicated if defs just kept major feature commit, save, rollback in netcore and netfx files.

### CODE REVIEW

**Review by johnnypham — APPROVED** [review]

File: src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlTransaction.cs [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+#if NETFRAMEWORK
+                && internalConnection.Is2005OrNewer
+#endif
+                && !_isFromAPI)
+            {
+                SqlClientEventSource.Log.TryAdvancedTraceEvent("SqlTransaction.Zombie | ADV | Object Id {0} yukon deferred zombie", ObjectID);
+            }
+            else
+            {
+                _internalTransaction = null; // pre-yukon zombification
```
Comment (DavoudEshtehari) [inline src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlTransaction.cs:123]:
```suggestion
                _internalTransaction = null; // pre SQL 2005 zombification
```

File: src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlTransaction.cs [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        ////////////////////////////////////////////////////////////////////////////////////////
+
+        private void ZombieCheck()
+        {
+            // If this transaction has been completed, throw exception since it is unusable.
+            if (IsZombied)
+            {
+                if (Is2005PartialZombie)
+                {
+                    _internalTransaction = null; // yukon zombification
```
Comment (DavoudEshtehari) [inline src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlTransaction.cs:138]:
```suggestion
                    _internalTransaction = null; // SQL 2005 zombification
```

**Review by DavoudEshtehari — COMMENTED** [review]
Could you move the `SqlTransaction.netcore` and `SqlTransaction.netfx` files back into their project's hierarchy?

**Review by DavoudEshtehari — APPROVED** [review]

File: src/Microsoft.Data.SqlClient/netcore/src/Microsoft.Data.SqlClient.csproj [hunk 3]
```diff
@@ -645,7 +648,7 @@
     <Compile Include="Microsoft\Data\SqlClient\SqlDelegatedTransaction.cs" />
     <Compile Include="Microsoft\Data\SqlClient\SqlInternalConnection.cs" />
     <Compile Include="Microsoft\Data\SqlClient\SqlInternalConnectionTds.cs" />
-    <Compile Include="Microsoft\Data\SqlClient\SqlTransaction.cs" />
+    <Compile Include="Microsoft\Data\SqlClient\SqlTransaction.cs"/>
```
Comment (David-Engel) [inline src/Microsoft.Data.SqlClient/netcore/src/Microsoft.Data.SqlClient.csproj:651]: Revert for consistency
```suggestion
    <Compile Include="Microsoft\Data\SqlClient\SqlTransaction.cs" />
```

**Review by David-Engel — APPROVED** [review]

**Review by JRahnama — APPROVED** [review]

**Pull request merged**